src/backend/distributed/commands/utility_hook.c

```diff
@@ -721,6 +725,13 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 		{
 			activeDropSchemaOrDBs--;
 		}
+
+
+		if (IsA(parsetree, ExplainStmt) &&
+			IsA(((ExplainStmt *) parsetree)->query, Query))
+		{
+			ExplainStatementRunning = false;
```

Comment:
I suppose we could technically have an explain in explain if it's an EXPLAIN ANALYZE that calls a function that does an explain. Usually it's better to use levels.